addons/mrp/tests/test_order.py

```diff
@@ -5022,6 +5022,17 @@ def test_workcenter_with_resource_calendar_from_another_company(self):
         with self.assertRaises(UserError):
             self.workcenter_1.resource_calendar_id, = resource_calendar
 
+    def test_workorder_without_product(self):
```

Comment:
It's better but now the test does not fail without your fix, so it's useless
You probably need to set the product after the new workorder